cli/internal/extract/extractMd.go

```diff
@@ -28,11 +37,95 @@ func extractMd(id string, sourceID string, rawData []byte, db *sqlite.Queries) e
 	}
 
 	// Extract/insert sections
-	err = extractSections(id, sourceID, extractedData, db)
+	err = extractSections(id, sourceID, extractedData, !options.DisablePurposeExtraction, purposeKey, db)
 	if err != nil {
 		slog.Debug("extract.extractMd could not extract sections", "error", err)
 		return err
 	}
 
 	return nil
 }
+
+// Extract purpose statement from a document. If not found return <blank>
+func extractMdDocumentPurpose(document string, purposeKey string) string {
+	key := purposeKey + ":"
+	parts := strings.Split(document, "\n")
+	// Only support frontmatter extraction if the document starts with frontmatter
+	if len(parts) >= 3 && strings.HasPrefix(parts[0], "---") {
+		inFrontmatter := true
+		for _, line := range parts[1:] {
+			trimmedLine := strings.TrimSpace(line)
+			if strings.HasPrefix(trimmedLine, "---") {
+				inFrontmatter = false
```

Comment:
nit: Is tracking `inFrontmatter` necessary? or it could it just be something like:
```suggestion
		for _, line := range parts[1:] {
			trimmedLine := strings.TrimSpace(line)
			if strings.HasPrefix(trimmedLine, "---") {
				break
```